src/utilities/astroFunctions.py

```diff
@@ -633,30 +658,89 @@ def Tisserand_P1_2_P2(r_planet_vec, v_inf_P1_vec, v_inf_P2_vec):
-    ra_max = r_planet_vec[l] / AU * 1.8
+    ra_min = r_planet_vec[0] / astroConstants.AU * 0.2
+    ra_max = r_planet_vec[l] / astroConstants.AU * 1.8
     plt.xlim([rp_min, rp_max])
     plt.ylim([ra_min, ra_max])
     plt.xlabel('Radius of Periapse [AU]')
     plt.ylabel('Radius of Apoapse [AU]')
-    plt.title(r'$v_{\infty}^{Earth} $ = '+ str(v_inf_P1_vec) +r' km/s \n'
-              r'$v_{\infty}^{Saturn} $ = '+ str(v_inf_P2_vec) +' km/s')
+    plt.title(r'$v_{\infty}^{Earth} $ = ' + str(v_inf_P1_vec) + r' km/s \n'
+              r'$v_{\infty}^{Saturn} $ = ' + str(v_inf_P2_vec) + ' km/s')
+
 
 def Tisserand_ESU():
-    r_planet_vec = np.array([a_E, a_Saturn, a_U])
+    r_planet_vec = np.array(
+        [astroConstants.SMA_EARTH, astroConstants.SMA_SATURN, astroConstants.SMA_URANUS])
     v_inf_E_vec = np.array([9, 11, 13, 15, 17])
     v_inf_S_vec = np.array([6, 7, 8, 9, 10])
     Tisserand_P1_2_P2(r_planet_vec, v_inf_E_vec, v_inf_S_vec)
-    TisserandPlot(a_E, v_inf_E_vec[2], 'cyan')
-    TisserandPlot(a_Saturn, v_inf_S_vec[3], 'cyan')
+    TisserandPlot(astroConstants.SMA_EARTH, v_inf_E_vec[2], 'cyan')
+    TisserandPlot(astroConstants.SMA_SATURN, v_inf_S_vec[3], 'cyan')
 
     plt.show()
 
+
+# ------------------------------------------------------------------------------------------------------------------- #
+#                                                DEPRECATED CONSTANTS
+#
+
+def __getattr__(name):
+    """ Special `__getattr__` to handle deprecation warning """
+    AU = 1.49597870691e8        # [km]
+    astroConstants = {
+        # Useful conversions
+        'R2D': 180.0 / np.pi,          # [deg]
+        'D2R': np.pi / 180.0,         # [rad]
+        'AU': AU,
+        'SEC2DAY': 1.0 / (3600*24),
+        'DAY2SEC': 1.0 * 3600 * 24,
+        'PI': np.pi,
+
+        # Gravitational parameters  [km^3 / s^2]
+        'mu_S': 1.32712440018e11,
+        'mu_E': 3.986004415e5,
+        'mu_M': 4.28283100e4,
+        'mu_V': 3.24858599e5,
+        'mu_J': 1.26686534e8,
+        'mu_Saturn': 3.79395000e7,
+        'mu_U': 5.79396566e6,
+        'mu_N': 6.83509920e6,
+        'mu_Moon': 4902.799,
+
+        # Planets' Radius   [km]
+        'S_radius': 695508,
+        'J_radius': 71492,
+        'V_radius': 6051.8,
+        'E_radius': 6378.1363,
+        'M_radius': 3396.19,
+        'Saturn_radius': 60268,
+        'U_radius': 25559,
+        'N_radius': 24764,
+        'Moon_radius': 1738.1,
+
+        # Planetary Heliocentric orbits     [km]
+        'a_E': 1.0 * AU,
+        'a_M': 1.52367934 * AU,
+        'a_Mercury': 0.387 * AU,
+        'a_V': 0.723 * AU,
+        'a_J': 5.203 * AU,
+        'a_Saturn': 9.537 * AU,
+        'a_U': 19.191 * AU,
+        'a_N': 30.069 * AU,
+        'a_P': 39.482 * AU,
+    }
+    if name in astroConstants:
+        deprecated.deprecationWarn(
+            name, '2025/08/31', 'Astro constants should be used from Basilisk.architecture.astroConstants')
+        return astroConstants[name]
+
```

Comment:
Nice way to provide prior functionality.  However, right now no code is checking that this deprecated functionality is working.  I drafted a unit test file that checks that all these variables are still available.  The deprecation warning is suppressed so we don't get false BSK deprecation warnings.  But, after the 1-year grace period is gone running purest will yield `BSKUrgentDeprecationWarning` which will be visible.  This will be an easy reminder to BSK developers to make a branch to remove this deprecated functionality.  I am attaching the test file I wrote.  You can include it as is, or edit if needed.  It goes into `src/utilities/tests`.
[test_legacyAstroConstants.py.txt](https://github.com/user-attachments/files/16931799/test_legacyAstroConstants.py.txt)

Reply:
Done